### ISSUE TITLE
FormSpec password fields use hidden text width, not "*" width

### ISSUE THREAD

**Issue by v-rob** [minetest/minetest#9368]
##### Minetest version
```
0.4.16 - 5.1.0
```

##### OS / Hardware

Operating system: Windows 10, Linux Mint

##### Summary/Steps to reproduce
In any password field (like the one for connecting to servers), type a bunch of wide letters (like `w`) until the characters start to scroll.  They will scroll long before you reach the end of the field as it is calculating the widths of the `w`s and predicts the text to be much longer than it actually is.  If you use a character like `|`, it will scroll later than it should because it predicts it to be shorter.

##### Solution
Calculate the width of the string in `*` characters, not normal characters.

**Comment by HybridDog** [comment]
I can confirm that this happens:

12 times w:
![12w](https://user-images.githubusercontent.com/3192173/74044753-dfca7e80-49cb-11ea-9ef6-62eedcf3f965.png)

12 times l:
![12l](https://user-images.githubusercontent.com/3192173/74044771-e6f18c80-49cb-11ea-9af8-c6d4f151dcd9.png)

**Comment by Zughy** [comment]
This is not a direct Minetest issue, what happens inside the password fields is completely managed [by Irrlicht
](http://irrlicht.sourceforge.net/docu/classirr_1_1gui_1_1_i_g_u_i_edit_box.html#a59a0cf8090eb165fdbade6144bc19807).

Related code:
https://github.com/minetest/minetest/blob/5fcc78a1feeffbc1a4fdd1cfffb49451694786d3/src/gui/guiFormSpecMenu.cpp#L1483-L1486

**Comment by v-rob** [comment]
True, but we own most of the EditBox code; we just need to completely unify it (nerzhul is doing so) and use that class everywhere.
